rclcpp/src/rclcpp/time_source.cpp

```diff
@@ -266,6 +286,10 @@ class TimeSource::NodeState final
       throw std::invalid_argument("Invalid type for parameter 'use_sim_time', should be 'bool'");
     }
 
+    on_set_parameters_callback_ = node_parameters_->add_on_set_parameters_callback(
+      std::bind(&TimeSource::NodeState::on_set_parameters, this, std::placeholders::_1));
```

Comment:
these changes look unrelated to adding clock_type to node_options, right?

Reply:
The callback is for handling illegal use_sim_time parameter set discussed [here](https://github.com/ros2/rclcpp/pull/1982#discussion_r978819552).